cmd/root.go

```diff
@@ -291,17 +291,40 @@ func readObjs(cmd *cobra.Command, paths []string, opts ...utils.ReadOption) ([]*
 		paths = append(paths, utils.ToDataURL(exec))
 	}
 
+	overlayCodeFile, err := flags.GetString(flagOverlayCodeFile)
+	if err != nil {
+		return nil, err
+	}
+
 	overlay, err := flags.GetString(flagOverlay)
 	if err != nil {
 		return nil, err
 	}
 	if overlay != "" {
+		log.Warnf("%q is deprecated, please use %q", flagOverlay, flagOverlayCodeFile)
```

Comment:
Is there perhaps some way to avoid duplicating this deprecation warning across every command?

Reply:
I am blind and should stop reviewing code tonight

Reply:
I am blind and should stop reviewing code for the night. This only occurs once.

Reply:
you made me spend 2m more to find the native way in pflag to deprecated a flag